### PR TITLE
feat: ButtonPopover and UserProfilePopover components

### PR DESCRIPTION
**Pull request by benschell-okta**
[OKTA-826088](https://oktainc.atlassian.net/browse/OKTA-826088)

## Summary

A few proposed new (Labs) components:
- ButtonPopover: uses a `<Button>` to trigger a MUI `<Popover>` to appear with some basic padding
- UserProfilePopover: Similarly, uses a (MUI) `<Button>` with a `<UserProfile>` as the contents to trigger a MUI `<Popover>` to appear with some basic padding
- Also UserProfile stories to storybook (alongside stories for each of the above). Added a new prop to UserProfile to have an icon next to the username (to have the v icon as pictured in Figma)

UserProfile needs additional styling, but that's true regardless of this new addition.

## Testing & Screenshots

- [ ] I have confirmed this change with my designer and the Odyssey Design Team.

ButtonPopover:
Screenshot 2024-11-04 at 2 36 21 PM

UserProfilePopover:
Screenshot 2024-11-04 at 2 36 41 PM

Practical usage in UIShell-based Admin:
Screenshot 2024-11-04 at 2 37 36 PM

Screenshot 2024-11-04 at 2 37 51 PM

### CODE REVIEW

File: packages/odyssey-react-mui/src/labs/TopNav/UserProfile.tsx [hunk 1]
```diff
@@ -37,6 +38,7 @@ const UserProfileIconContainer = styled("div", {
 const UserProfileInfoContainer = styled("div")(() => ({
   display: "flex",
   flexDirection: "column",
+  textAlign: "left",
```
Comment (bryancunningham-okta) [inline packages/odyssey-react-mui/src/labs/TopNav/UserProfile.tsx:41]: Doesn't really seem like we need any of these styles. Fairly certain we can remove this styled component entirely and render either a plain `div` or a fragment

File: packages/odyssey-react-mui/src/labs/ButtonPopover.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+   * The horizontal alignment of the popover.
+   */
+  popoverAlignment?: PopoverProps["popoverAlignment"];
+  /**
+   * The contents of the popover
+   */
+  children: ReactNode | NullElement;
+} & ButtonProps;
+
+const ButtonPopover = (props: ButtonPopoverProps) => {
```
Comment (bryancunningham-okta) [inline packages/odyssey-react-mui/src/labs/ButtonPopover.tsx:38]: Can we name this something like... `ButtonWithPopover` or something similar? Just something that more accurately indicates what this component is/does? The current name leads me to believe that this component is *just* the popover. But, maybe that's just me. Any thoughts?

File: packages/odyssey-react-mui/src/labs/ButtonPopover.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  }, []);
+  const popoverClosed = useCallback(() => {
+    setIsOpen(false);
+  }, []);
+
+  return (
+    <>
+      <Button
+        {...buttonProps}
+        isDisabled={buttonProps.isDisabled || isOpen}
```
[the remark this answers is not in the document]
Comment (benschell-okta) [reply packages/odyssey-react-mui/src/labs/ButtonPopover.tsx:63]: The MUI Popover handles closing (it handles an overlay and clicking anywhere outside of the popover closes it). Kinda like Modal IIRC. Popover has an onClose handler to let you know when it has closed.

File: packages/odyssey-react-mui/src/labs/TopNav/UserProfile.tsx (hunk 1 above)
Comment (benschell-okta) [reply packages/odyssey-react-mui/src/labs/TopNav/UserProfile.tsx:41]: The `textAlign` is definitely needed (the default styles for `<button>` have `text-align: center`). As for the rest 🤷 ; the UserProfile component itself needs work more generally, I was just making modifications where necessary to get this Popover to work.

File: packages/odyssey-react-mui/src/labs/ButtonPopover.tsx (hunk 2 above)
Comment (benschell-okta) [reply packages/odyssey-react-mui/src/labs/ButtonPopover.tsx:38]: I have zero opinions and am happy to name it whatever. This PR Is mainly to start a conversation (and get design blessing)

Comment (bryancunningham-okta) [reply packages/odyssey-react-mui/src/labs/ButtonPopover.tsx:38]: Got ya, sounds good. Let's go with adaptable `ButtonWithPopover` for now then, if you don't mind :)

File: packages/odyssey-react-mui/src/labs/ButtonPopover.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (bryancunningham-okta) [reply packages/odyssey-react-mui/src/labs/ButtonPopover.tsx:63]: I don't really feel like we really gain anything by disabling this button when the menu is open do we? Feels like we may as well just leave it enabled. Was there a certain issue you ran into that this was solving?

[the remark this answers is not in the document]
Comment (benschell-okta) [reply packages/odyssey-react-mui/src/labs/ButtonPopover.tsx:63]: Not functionally, no. This was purely around the presentation, specifically this type of view: https://private-user-images.githubusercontent.com/116028555/382894810-99f3a5be-3e71-47b5-abb6-a648c729e2c6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA3NTA5NDQsIm5iZiI6MTczMDc1MDY0NCwicGF0aCI6Ii8xMTYwMjg1NTUvMzgyODk0ODEwLTk5ZjNhNWJlLTNlNzEtNDdiNS1hYmI2LWE2NDhjNzI5ZTJjNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNFQyMDA0MDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MGRmM2Q3NjI0ZjU5NDViZjcxZDA1ZWM1ZmQ2ODcwMmE2MzU3MmI1Y2UyMWEyOGJkM2E4YzdjZGMyMWFiNDViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.BArm2CApDOvnKyjNI4FGHz5ZcOokrGuKQZARaZrTaHQ

The UserProfile piece has no border or anything when not hovered. There is no design in Figma that shows how it should change when interactive (hovering/active), so I just took a guess: I figured we'd want something like the disabled view. So what I did here is what you can see in that screenshot.

File: packages/odyssey-react-mui/src/labs/TopNav/UserProfile.tsx (hunk 1 above)
Comment (bryancunningham-okta) [reply packages/odyssey-react-mui/src/labs/TopNav/UserProfile.tsx:41]: No worries. Just figured I'd flag it. If you need the `text-align` for a button, I'd probably just pull that button into it's own styled component and apply this there. That way it's a bit more clear what is happening/why this declaration is here.

Comment (benschell-okta) [reply packages/odyssey-react-mui/src/labs/TopNav/UserProfile.tsx:41]: I think I wasn't very clear in my description. 

In UserProfilePopover, I was wrapping this component with a MuiButton ([here](https://github.com/okta/odyssey/pull/2406/files#diff-4247df56544a88b974863fbda959827f387802aed697c828f00ac668343122b2R98)). In that scenario, a `text-align` from the button is causing a different layout here in the UserProfile. So this is to reset that / enforce the intended design of this component.

File: packages/odyssey-react-mui/src/labs/TopNav/UserProfilePopover.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  }, []);
+  const popoverClosed = useCallback(() => {
+    setIsOpen(false);
+  }, []);
+
+  return (
+    <>
+      <MuiButton
+        {...muiProps}
+        // aria-controls={ariaControls}
```
Comment (benschell-okta) [inline packages/odyssey-react-mui/src/labs/TopNav/UserProfilePopover.tsx:69]: I copied this from `<Button>` 's impl. I wasn't clear if / how many of these types of things should be set on something using `<MuiButton>`, so I commented them as a reminder to discuss this. Probably mostly a q for @KevinGhadyani-Okta

File: packages/odyssey-react-mui/src/labs/TopNav/UserProfile.tsx (hunk 1 above)
Comment (bryancunningham-okta) [reply packages/odyssey-react-mui/src/labs/TopNav/UserProfile.tsx:41]: Thanks for the extra context, I seen where you were wrapping this in a button. I was trying to say we should style that^ button with `textAlign: "left"` directly, rather than applying it here. The reasoning for the text align declaration would be more clear in that scenario, I think

Comment (benschell-okta) [reply packages/odyssey-react-mui/src/labs/TopNav/UserProfile.tsx:41]: I think my point, though, is that the UserProfile component should always be left aligned regardless of any container it is put in. In that case, it should ensure that it always has the correct alignment no matter what, which is why I'd advocate for having it here at a minimum. Basically, I'm suggesting we insulate this from any consumer doing something silly since this component is exported from Labs by itself as well.

tl;dr: what if someone did `<center><UserProfile /></center>`? We'd want the UserProfile component to render with proper alignment no matter what.

**Comment by benschell-okta** [comment]
Closing in favor of the new #2414

File: packages/odyssey-react-mui/src/Buttons/BaseButton.tsx [hunk 5]
```diff
@@ -225,7 +204,7 @@ const Button = ({
           type={type}
           variant={variant}
         >
-          {label}
+          {children ?? label}
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/Buttons/BaseButton.tsx:207]: Not sure why we're adding the `children` prop here.

File: packages/odyssey-react-mui/src/Buttons/BaseMenuButton.tsx [hunk 6]
```diff
@@ -174,31 +216,52 @@ const MenuButton = ({
...
+          anchorEl={anchorEl}
+          onClose={closeMenu}
+          anchorOrigin={anchorOrigin}
+          transformOrigin={transformOrigin}
+        >
+          <Box
+            sx={{
+              padding: odysseyDesignTokens.Spacing4,
+            }}
+          >
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/Buttons/BaseMenuButton.tsx:255]: We should make sure these styles are memoized.

Comment (benschell-okta) [reply packages/odyssey-react-mui/src/Buttons/BaseMenuButton.tsx:255]: @KevinGhadyani-Okta this is the initial pass and has been closed. Please look at #2414 instead.
